spec/datadog/core/configuration/settings_spec.rb

```diff
@@ -707,7 +708,24 @@
             context "is defined as #{value}" do
               let(:environment) { value.to_s }
 
-              before { expect(Datadog::Core).to receive(:log_deprecation) }
+              context 'log deprecation when deprecated env var is set' do
+                let(:message_matcher) { /DD_PROFILING_PREVIEW_GVL_ENABLED/ }
+
+                before do
+                  # This is initialized during app startup and should not change during app lifecycle
+                  # However in our tests, we change the environment variables without completely resetting the app
+                  # This is why we reset this variable here.
+                  Datadog::Core::Configuration::ConfigHelper.instance_variable_set(:@log_deprecations_called_with, nil)
+                end
+
+                it 'logs deprecation' do
+                  expect(Datadog::Core).to receive(:log_deprecation).with(any_args) do |&message_block|
+                    expect(message_block.call).to match(message_matcher) if message_matcher
+                  end
+                  # Reinitialize components (which logs deprecations again as we've reset @log_deprecations_called_with)
+                  Datadog.configure {}
+                end
+              end
```

Comment:
TBH: Feel free to remove this. I'm happy with the generic coverage added in this PR for logging deprecations and I don't think we need extra coverage here.